data_steward/cdr_cleaner/clean_cdr.py

```diff
@@ -77,15 +78,16 @@ def _gather_ehr_rdr_queries(project_id, dataset_id):
     query_list.extend(replace_standard_concept_ids.replace_standard_id_in_domain_tables(project_id, dataset_id))
     query_list.extend(domain_alignment.domain_alignment(project_id, dataset_id))
     query_list.extend(id_dedup.get_id_deduplicate_queries(project_id, dataset_id))
-    query_list.extend(null_foreign_key.null_invalid_foreign_keys(project_id, dataset_id))
     query_list.extend(clean_years.get_year_of_birth_queries(project_id, dataset_id))
     query_list.extend(neg_ages.get_negative_ages_queries(project_id, dataset_id))
     query_list.extend(bad_end_dates.get_bad_end_date_queries(project_id, dataset_id))
     query_list.extend(no_data_30days_after_death.no_data_30_days_after_death(project_id, dataset_id))
     query_list.extend(valid_death_dates.get_valid_death_date_queries(project_id, dataset_id))
     query_list.extend(drug_refills_supply.get_days_supply_refills_queries(project_id, dataset_id))
     query_list.extend(populate_routes.get_route_mapping_queries(project_id, dataset_id))
+    query_list.extend(fix_datetimes.get_fix_incorrect_datetime_to_date_queries(project_id, dataset_id))
     query_list.extend(remove_records_with_wrong_date.get_remove_records_with_wrong_date_queries(project_id, dataset_id))
+    query_list.extend(null_foreign_key.null_invalid_foreign_keys(project_id, dataset_id))
```

Comment:
@nishanthpp93 why do we move null_foreign_key to the end?

Reply:
This is because some rules above can create invalid foreign keys, which need to be cleaned.